bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/thing/ThingResource.java

```diff
@@ -461,6 +463,50 @@ public Response getStatus(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) String langu
         return Response.ok(null, MediaType.TEXT_PLAIN).entity(thingStatusInfo).build();
     }
 
+    @PUT
+    @RolesAllowed({ Role.USER, Role.ADMIN })
+    @Path("/{thingUID}/enable")
+    @ApiOperation(value = "Disables a thing.")
+
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK", response = String.class),
+            @ApiResponse(code = 400, message = "Thing could not be disabled as it does not have a handler."),
+            @ApiResponse(code = 404, message = "Thing not found."),
+            @ApiResponse(code = 409, message = "Thing could not be disabled as it is not editable.") })
+
+    public Response setEnabled(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) String language,
+            @PathParam("thingUID") @ApiParam(value = "thing") String thingUID,
+            @ApiParam(value = "enabled") String enabled) throws IOException {
+        final Locale locale = LocaleUtil.getLocale(language);
+
+        ThingUID thingUIDObject = new ThingUID(thingUID);
+
+        // Check if the Thing exists, 404 if not
+        Thing thing = thingRegistry.get(thingUIDObject);
+        if (null == thing) {
+            logger.info("Received HTTP PUT request for disable thing at '{}' for the unknown thing '{}'.",
+                    uriInfo.getPath(), thingUID);
+            return getThingNotFoundResponse(thingUID);
+        }
+
+        // ask whether the Thing exists as a managed thing, so it can get
```

Comment:
Why this check? We can also disable unmanaged things as this information is not persisted with the Thing entities!